R/tbl_svysummary.R

```diff
@@ -0,0 +1,470 @@
+#' It is similar to [tbl_summary()].
+#'
+#' @param data A survey object created with created with [survey::svydesign()]
+#' @inheritParams tbl_summary
+#'
+#' @details
+#' See [tbl_summary()] for more details on the different arguments.
+#'
+#' @section statistic argument:
+#' The statistic argument specifies the statistics presented in the table. The
+#' input is a list of formulas that specify the statistics to report. For example,
+#' `statistic = list(age ~ "{mean} ({sd})")` would report the mean and
+#' standard deviation for age; `statistic = list(all_continuous() ~ "{mean} ({sd})")`
+#' would report the mean and standard deviation for all continuous variables.
+#'  A statistic name that appears between curly brackets
+#' will be replaced with the numeric statistic (see [glue::glue]).
+#'
+#' For categorical variables the following statistics are available to display.
+#' \itemize{
+#'   \item `{n}` frequency
+#'   \item `{N}` denominator, or cohort size
+#'   \item `{p}` formatted percentage
+#'   \item `{n_unweighted}` unweighted frequency
+#'   \item `{N_unweighted}` unweighted denominator
+#'   \item `{p_unweighted}` unweighted formatted percentage
+#' }
+#' For continuous variables the following statistics are available to display.
+#' \itemize{
+#'   \item `{median}` median
+#'   \item `{mean}` mean
+#'   \item `{sd}` standard deviation
+#'   \item `{var}` variance
+#'   \item `{min}` minimum
+#'   \item `{max}` maximum
+#'   \item `{p##}` any integer percentile, where `##` is an integer from 0 to 100
+#' }
+#'
+#' Unlike [tbl_summary()], It's not possible to pass a custom function.
+#'
+#' For both categorical and continuous variables, statistics on the number of
+#' missing and non-missing observations and their proportions are available to
+#' display.
+#' \itemize{
+#'   \item `{N_obs}` total number of observations
+#'   \item `{N_miss}` number of missing observations
+#'   \item `{N_nonmiss}` number of non-missing observations
+#'   \item `{p_miss}` percentage of observations missing
+#'   \item `{p_nonmiss}` percentage of observations not missing
+#'   \item `{N_obs_unweighted}` unweighted total number of observations
+#'   \item `{N_miss_unweighted}` unweighted number of missing observations
+#'   \item `{N_nonmiss_unweighted}` unweighted number of non-missing observations
+#'   \item `{p_miss_unweighted}` unweighted percentage of observations missing
+#'   \item `{p_nonmiss_unweighted}` unweighted percentage of observations not missing
+#' }
+#'
+#' Note that for categorical variables, `{N_obs}`, `{N_miss}` and `{N_nonmiss}` refer
+#' to the total number, number missing and number non missing observations
+#' in the denominator, not at each level of the categorical variable.
+#' @export
+#' @importFrom stats as.formula weights
+#' @return A `tbl_svysummary` object
+#' @family tbl_svysummary tools
+#' @author Joseph Larmarange
+#' @examples
+#' # A simple weighted dataset
+#' tbl_svysummary_ex1 <-
```

Comment:
Maybe it could be an option to consider when `include = NULL` and therefore performed only if the user did not specify a list on its own.

In such case, we could have a look on the `call` element of the survey object and have a look to the following parameters: `id`, `probs`, `strata`, `fpc` and `weights`. All these variables should be excluded by default.

I just committed a change to implement it. cf https://github.com/ddsjoberg/gtsummary/pull/547/commits/7d9d8d32117d618db6ad347feed956822f5bfecf

Reply:
The example works within the `include=` from the titanic dataset, but the clustered data produces an error

```r
data(api, package = "survey")
d_clust <- survey::svydesign(id = ~dnum, weights = ~pw, data = apiclus1, fpc = ~fpc)
tbl_svysummary(d_clust, by = both) 

Error: Problem with `mutate()` input `df_stats`.
x error in evaluating the argument 'x' in selecting a method for function 't': need at least two non-NA values to interpolate
i Input `df_stats` is `pmap(...)`.
Run `rlang::last_error()` to see where the error occurred. 
```

Reply:
OK the problem is due in fact to the flag variable which is equal to NA for all rows. 

a8467b23e5be92bfa21c8894d0a013d7f9fa071c should fix it